ci/build_wheel_cuspatial.sh

```diff
@@ -3,4 +3,6 @@
 
 set -euo pipefail
 
+export SKBUILD_CMAKE_ARGS="-DUSE_LIBARROW_FROM_PYARROW=ON"
```

Comment:
What if someone is building a wheel on their own? Don't they also need to pass this flag? Why can't we detect this in CMake?

Reply:
See [this comment](https://github.com/rapidsai/cudf/pull/14565#discussion_r1420589884) from @robertmaynard and [this commit](https://github.com/rapidsai/cudf/pull/14565/commits/e8a90becc91b8a380c168fa4ada5cacd781bc584) showing what I was previously doing for autodetection before removing it.

Not sure if that's the kind of detection that you're imagining.